### PR TITLE
Test: Delete all with Timeout.

### PR DESCRIPTION
**Pull request by eloycoto**
Had seen in the nightly builds that the delete all has failed and hang
until the Jenkins timeout. Tested locally and pods were all deleted but
the kubectl was still hung.

With this change we delete all resources with a timeout.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5056)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-nightly

**Comment by ianvernon** [comment]
I feel like this is masking over the actual problem. A call to delete should not take more than 300 seconds. Are there extra processes left running on the nodes that we do not know about, causing extreme load, etc.?

**Comment by eloycoto** [comment]
I tested and no, happens a few of times in my local env too but the CI is blocked a lot because of this.

I'm sure that it's a problem in there, but due is the only place that happens, and that I need to have something that does not provide any timeout in the CI to gather logs to continue to debug what is the real problem :-(

**Comment by ianvernon** [comment]
@eloycoto fair enough, better to timeout than to deadlock and have no logs.

**Review by ianvernon — APPROVED** [review]

**Comment by ianvernon** [comment]
@eloycoto is there any chance there is a leak of stuff in the nightly test framework? The amount of goroutines that are spinning is astronomical in the failed build here: 
https://jenkins.cilium.io/job/Cilium-PR-Nightly-Tests-All/165/

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
Hi @ianvernon 

I think that this PR can be merged, the nightly test are not yet fixed, but at
least we don't have the timeout. The reason of the timeout, is that `kubectl
delete --all pod,services,cnp` will delete service kubernetes and it breaks how
kubectl behaves (Change in kubernetes 1.11).

That issue is fixed on PoliciesNightly and in the Nightly I need to make a bit
more changes, but I would like to clean the timeouts and continue to work in
Nightly in meanwhile.

Thanks

File: test/helpers/policygen/models.go [hunk 1]
```diff
@@ -529,7 +536,9 @@ func (t *TestSpec) CreateCiliumNetworkPolicy() (string, error) {
 	  "kind": "CiliumNetworkPolicy",
 	  "metadata": {
 		"name": "%[1]s",
-		"test": "%[3]s"
+		"labels": {
+			"test": "policygen"
+		}
```
Comment (raybejjani) [inline test/helpers/policygen/models.go:541]: Do we need to also edit https://github.com/eloycoto/cilium/blob/5d28eac43263a05344b7e974214c4ac77dcc3543/test/helpers/policygen/models.go#L638 to match the number of `%*` here?

**Review by raybejjani — COMMENTED** [review]
Is there any chance we can edit the log at https://github.com/eloycoto/cilium/blob/5d28eac43263a05344b7e974214c4ac77dcc3543/test/k8sT/Nightly.go#L69 to more clearly say it was part of a `deleteAll` ?
Relatedly, it might be nice to print out the existing things in k8s, when we see this failure,  in case there is a trend of some sort (I think `kubectl get -o wide all --all-namespaces` prints most things).

File: test/helpers/policygen/models.go (hunk 1 above)
Comment (eloycoto) [reply test/helpers/policygen/models.go]: Nope, it's not needed at all ;-)

Comment (eloycoto) [reply test/helpers/policygen/models.go:541]: Done.

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
Another trigger of the build. Please @ianvernon if you can review again it'll be nice.

File: test/helpers/policygen/models.go [hunk 2]
```diff
@@ -529,7 +536,9 @@ func (t *TestSpec) CreateCiliumNetworkPolicy() (string, error) {
 	  "kind": "CiliumNetworkPolicy",
 	  "metadata": {
 		"name": "%[1]s",
-		"test": "%[3]s"
+		"labels": {
```
Comment (ianvernon) [inline test/helpers/policygen/models.go:539]: What is the reason for these changes? Is this because the metadata field does not accept arbitrary key names? Please provide more context.

**Review by ianvernon — APPROVED** [review]

File: test/helpers/policygen/models.go (hunk 2 above)
Comment (eloycoto) [reply test/helpers/policygen/models.go:539]: Test label was not in place at all because it was not into labels.
Now we use the label test to set to policygen to be able to delete all by `-l test=policygen`
